### ISSUE TITLE
RFC: add tabindex=0 to #pane-preview for keyboard users

### ISSUE THREAD

**Issue by phiw13** [textpattern/textpattern#2080]
For accessibility reason, to make the content of the preview-panel scrollable with the keyboard, set tabindex=0 to the div#pane-preview.

Users who `tab` through the controls at the top of preview pane can then scroll through the article preview. At present it kinda work (only) when there is a focus-able element in the article body (e.g. a link). But no such option for the HTML view.

I think, adding the `tabindex=0` on line 413 of txp_article.php is enough (that works for me in a simple local test).

Thank you.

**Comment by phiw13** [comment]
Note for the `iframe` preview it appears more complicated as, at the moment, the code adds `tabindex=-1` to disable setting focus.

**Comment by Bloke** [comment]
@phiw13 Like that? [1f618dc](https://github.com/textpattern/textpattern/commit/1f618dc0befc88287e3b4b0ff99a709d72c56c4d)

> Note for the `iframe` preview it appears more complicated

Hmmm. Not sure how to tackle that. @bloatware any ideas?

**Comment by phiw13** [comment]
For the inline pane (`#pane-preview`), that is perfect, thank you.

**Comment by bloatware** [comment]
I (or is it jQuery UI?) must have added `tabindex="-1"` for some reason, but do not remember why. Probably, to avoid admin/public tab crossing, which looked reasonable for me. But I do not mind if you prefer to remove it.

**Comment by phiw13** [comment]
> But I do not mind if you prefer to remove it.

In that case, would you min setting it to `tabindex="0"`. That would have the iframe in the chain for keyboard users and make it scrollable even it contains no focusable elements.

TY.

**Comment by bloatware** [comment]
No worries, but this would also make tab jump through all focusable elements of the article. This can be disturbing for the normal admin side workflow, especially if the article contains many focusable elements..

Maybe, we can enable this 'focus inside preview' action via some external button/link? Those who wish, could then enter the frame, and the others would just skip this button. Looks a bit complicated, ideas welcome.

**Comment by phiw13** [comment]
Is that `tabindex="-1"` not set on the (that) `#preview-frame` directly? If I read txp_article.php correctly, line 853…

That would not affect anything but that iframe, I think.

But as I think about it, simply removing that `tabindex="-1"` from the `#preview-frame` iframe would also solve the issue, as the content of the iframe will always have some type of focusable element (it is a full view of the website after all)

**Comment by petecooper** [comment]
Room read: are we close for this to be ready / done in 4.9.2, or is 4.9.3 more realistic? Thank you.

**Comment by bloatware** [comment]
I can certainly remove `tabindex`, then caveat utilitor.
